pkg/kubelet/kuberuntime/security_context.go

```diff
@@ -34,11 +36,13 @@ func (m *kubeGenericRuntimeManager) determineEffectiveSecurityContext(pod *v1.Po
 		}
 	}
 
+	useSeccompPerDefault := utilfeature.DefaultFeatureGate.Enabled(features.SeccompDefault)
```

Comment:
An option may be to add a kubelet flag that allows one to pick runtime/default, unconfined or custom. A similar idea was captured as an alternative in the KEP.

Reply:
Feature gates are intended to cover the development life cycle of a feature - they are not intended to be long-term APIs. If we expect admins to have kubelet-level control over this once this feature graduates to GA, this needs a dedicated kubelet configuration option

Reply:
> Feature gates are intended to cover the development life cycle of a feature - they are not intended to be long-term APIs.

Agree.

> If we expect admins to have kubelet-level control over this once this feature graduates to GA, this needs a dedicated kubelet configuration option.

Yes. I brought up the alternative if there are concerns that there is no opt out at a global level in the kubelet once this goes GA. 
I still think that it is fine if this is enabled by default when we go to GA. The users opt out of seccomp at the pod level as needed. 

cc: @tallclair

Reply:
> I still think that it is fine if this is enabled by default when we go to GA. The users opt out of seccomp at the pod level as needed.

How do users on a managed cluster get that signal other than noticing they are broken on an upgrade to a version where this is enabled?

Reply:
Note that both cri-o and containerd let you configure the runtime default. IMO the configuration of the default should be left to the runtime, and Kubelet should (once this goes to GA) always defer to the runtime for the default, rather than making this configurable in the Kubelet.

Reply:
> How do users on a managed cluster get that signal other than noticing they are broken on an upgrade to a version where this is enabled?

Thinking out aloud on how a managed service would roll out this out if there was a kubelet configuration option for this: 
Would they ask the users to opt-in to this feature as a cluster option?
If yes, how would the user know if they can safely do it?
They would have to first test their pods by enabling runtime/default on them (on older unupgraded cluster) and see if they work or set them to unconfined explicitly or use some custom policy for their workloads. I don't see how they can get by without testing and looking at logs.
If they are comfortable with this test, they will then go ahead with the update removing the runtime/default from the pod and rely on the cluster configuration for runtime/default instead.
If we go with this model, we will have some users that never opt-in to this feature because they don't want to test it or don't have the time to test till a CVE motivates them to do it.
This approach works, but doesn't improve the security by default. 

The approach in this PR is taking us on a path to forcing everyone to use seccomp by default.

So, we need to pick whether we want to force everyone to use seccomp by default or allow them to not use it as long as they want to. 

Even with force approach, someone can still craft a runtime default seccomp policy which is close to unconfined.

Reply:
> The approach in this PR is taking us on a path to forcing everyone to use seccomp by default.

it seems like it's changing the meaning of an empty seccomp field in the Pod API... how is that not a breaking change?

Reply:
> it seems like it's changing the meaning of an empty seccomp field in the Pod API... how is that not a breaking change?

Yes, it is. So, what would be the preferred way to introduce this? Is the kubelet configuration option acceptable or something else?

Reply:
> Is the kubelet configuration option acceptable or something else?

That's what I expected.

Reply:
> That's what I expected.

Sounds good to me. @tallclair @saschagrunert @derekwaynecarr wdyt?

Reply:
I'm fine with that, too. As a summary what we now do:

- Replace the feature gate with a kubelet configuration option (CLI and config file)
- Update the KEP to reflect that change
- Reconsider alpha to beta upgrades within the KEP

Is this correct?

Reply:
I don't think it's a good idea to release a feature that will break workloads on upgrade. We should require a flag to enable this even in GA.

Reply:
Added a `SeccompDefault` kubelet configuration as well as a corresponding CLI flag `--seccomp-default`. Both are now covered by the feature gate check and everything is disabled by default.